apps/kyb-app/src/components/layouts/AppShell/Logo.tsx

```diff
@@ -16,14 +17,16 @@ const prefetchImage = (url: string) =>
 const fallback = (timeout: number) => new Promise(resolve => setTimeout(resolve, timeout));
 
 export const Logo = ({ logoSrc, appName, onLoad }: Props) => {
+  const onLoadRef = useRefValue(onLoad);
+
   useEffect(() => {
     if (!onLoad) {
       return;
     }
 
     // Using race here in case if image is corrupted or load takes to long we don't want to lock stepper breadcrumbs forever.
-    Promise.race([prefetchImage(logoSrc), fallback(3000)]).then(onLoad);
-  }, [logoSrc, onLoad]);
+    Promise.race([prefetchImage(logoSrc), fallback(3000)]).then(onLoadRef.current);
+  }, [logoSrc, onLoadRef]);
```

Comment:
Wouldn't that bring back the bug ? I think best way to handle would be something like

```suggestion
    Promise.race([prefetchImage(logoSrc), fallback(3000)]).then((value) => { if (onLoadRef.current) onLoadRef.current(value)) });
  }, [logoSrc, onLoadRef]);
```
(cleanup is needed, just an idea suggestion)

Reply:
Nope. It solves bug. Ref object is same each rerender whenever current is being updated. What you suggesting is pretty much safe check but its not needed in case with`then` since it wont do anything if it will recieve `undefined`

Reply:
I was referring to proposed above changes in the first half :)

This isn't just a safe-check - onLoadRef will be updated with latest onLoad callback, and this will make sure that we always call most-recent version of the callback ( and thats something that I would expect from the callbacks ).